### PR TITLE
feat: Merge ethexe implementation

### PR DESCRIPTION
**Pull request by NikVolf**
don't squash!

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -75,6 +75,11 @@ jobs:
       - name: "Install: Rust toolchain"
         uses: dsherret/rust-toolchain-file@v1
 
+      - name: "Install: Foundry"
```
Comment (StackOverflowExcept1on) [inline .github/workflows/build.yml:78]: it would be nice to check solidity code on CI (see #4059)

Comment (NikVolf) [reply .github/workflows/build.yml:78]: Add it later

**Review by grishasobol — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -5,7 +5,7 @@ target-xwin/
 log/
 weight-dumps/
 .binpath
-.vscode
+/.vscode
```
Comment (breathx) [inline .gitignore:8]:
```suggestion
.vscode
```

File: ethexe/cli/src/chain_spec.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    /// Address of Ethereum Router contract.
+    pub ethereum_router_address: String,
+
+    /// Addresses of Boot Nodes.
+    pub bootnodes: Vec<Multiaddr>,
+}
+
+pub fn testnet_config() -> ChainSpec {
+    ChainSpec {
+        ethereum_router_address: "0x05069E9045Ca0D2B72840c6A21C7bE588E02089A".to_string(),
```
Comment (StackOverflowExcept1on) [inline ethexe/cli/src/chain_spec.rs:17]: `testnet_config()` is used when the cli option `--chain-spec` is set to `dev`
```suggestion
        ethereum_router_address: "0xf90d2956B6F36194fbac181Fb95b2f67274821da".to_string(),
```

Comment (NikVolf) [reply ethexe/cli/src/chain_spec.rs:17]: Update it once merged, it's not a purpose of this PR
It's pure master from other repo

File: Makefile [hunk 4]
```diff
@@ -1,3 +1,22 @@
+# Temporary section for eGPU.
+.PHONY: gpu-pre-commit
+gpu-pre-commit:
+	@ echo " > Formatting eGPU" && cargo +nightly fmt --all -- --config imports_granularity=Crate,edition=2021
+	@ echo " >> Clippy checking eGPU" && cargo clippy -p "ethexe-*" --all-targets --all-features -- --no-deps -D warnings
+
+# Building contracts
+.PHONY: gpu-contracts-pre-commit
+gpu-contracts-pre-commit:
```
Comment (breathx) [inline Makefile:9]: Idk should we, but: 
```suggestion
# Ethexe section.
.PHONY: ethexe-pre-commit
ethexe-pre-commit:
	@ echo " > Formatting ethexe" && cargo +nightly fmt --all -- --config imports_granularity=Crate,edition=2021
	@ echo " >> Clippy checking ethexe" && cargo clippy -p "ethexe-*" --all-targets --all-features -- --no-deps -D warnings

# Building ethexe contracts
.PHONY: ethexe-contracts-pre-commit
ethexe-contracts-pre-commit:
```

**Review by breathx — APPROVED** [review]

**Pull request merged**